content/sips/sip-366.md

```diff
@@ -16,36 +16,277 @@ proposal: >-
+    address collateralType,
+    uint256 collateralAmountD18,
+    uint256 leverage
+);
+```
+
+#### Submitting Intents
+
+Add the following function to **declare** and record an intent to delegate a new amount of collateral (refer to the NatSpec annotations for further implementation guidance):
+
+```solidity
+/**
+ * @notice Declare an intent to delegate a new amount of collateral
+ * @dev The {msg.sender} is not necessarily the account owner but it
+ * must be authorized to act on behalf of the account
+ * @param accountId The ID of the account intending to delegate a new
+ * amount of collateral
+ * @param poolId The ID of the pool to which the account intends to
+ * delegate a new amount of collateral
+ * @param collateralType The address of the collateral type that the
+ * account intends to delegate
+ * @param newCollateralAmountD18 The new amount of collateral that
+ * the account intends to delegate, denominated with 18 decimals of
+ * precision
+ * @param leverage The new intended leverage used in the position
+ */
+function declareIntentToDelegateCollateral(
+    uint128 accountId,
+    uint128 poolId,
+    address collateralType,
+    uint256 newCollateralAmountD18,
+    uint256 leverage
+) public;
+```
+
+The `declareIntentToDelegateCollateral` function should, at minimum, satisfy the following requirements:
+
+1. Ensure the caller is authorized to represent the account.
+2. Verify the account holds enough collateral to execute the intent.
+3. Check the validity of the collateral amount to be delegated, considering the `escrowedCollateralByAccount`.
+4. Update the `escrowedCollateralByAccount` to reflect the collateral amount declared in the intent for the identified account and pool.
+5. Update the `intentNoncesByAccount` to include the new intent nonce associated with the account.
+6. Update the `intentByNonce` to represent the newly declared intent using the specific nonce.
+7. Emit a `DelegateCollateralIntentDeclared` event.
+
+#### Processing Intents
+
+Add the following function to **process** an outstanding intent to delegate a new amount of collateral (refer to the NatSpec annotations for further implementation guidance):
+
+```solidity
+/**
+ * @notice Attempt to process an outstanding intent to delegate a
+ * new amount of collateral
+ * @dev The {msg.sender} can be any address
+ * @param accountId The ID of the account for which the outstanding
+ * intent to delegate a new amount of collateral is being processed
+ * @param poolId The ID of the pool for which the intent of the account
+ * to delegate a new amount of collateral is being processed
+ * @param collateralType The address of the collateral type that the
+ * account has an outstanding intent to delegate a new amount of
+ */
+function processIntentToDelegateCollateral(
+    uint128 accountId,
+    uint128 poolId,
+    address collateralType
+) public;
+```
+
+The `processIntentToDelegateCollateral` function should, at minimum, satisfy the following requirements:
```

Comment:
Probably we need the workflow for undelegation, because in that case, we need to check that what the user has delegated is greater than or equal to `undelegateCollateralCachePerAccountPerPool`

Reply:
for the sake of brevity, I reworked this section to clearly define expectations while leaving out the growing number of logic checks. line must be drawn somewhere